src/oscar/apps/basket/forms.py

```diff
@@ -19,6 +19,10 @@ def __init__(self, strategy, *args, **kwargs):
         super(BasketLineForm, self).__init__(*args, **kwargs)
         self.instance.strategy = strategy
 
+        num_available = getattr(self.instance.purchase_info.availability, 'num_available', None)
+        max_allowed_quantity = min(num_available, self.instance.basket.max_allowed_quantity()[0])
```

Comment:
This will give an incorrect result (`None`) if `num_available` is `None`. Is there a reason you're using `getattr` above? The availability class is required to set `num_available` always.

Reply:
The reason is that num_available required only by StockRequired availability class, others don't require this argument.

Reply:
But the result of `min(None, 6)` will be `None`, meaning `max_allowed_quantity` will be set to `None`, which I don't think is what you want?